Dockerfile

```diff
@@ -0,0 +1,45 @@
+FROM ocaml/opam:debian-10-ocaml-4.11
+
+LABEL maintainer="gracewgao"
+
+# installs system dependencies
+RUN sudo apt update && sudo apt upgrade -y \
+    && sudo apt install python3 python3-pip pkg-config libsqlite3-dev build-essential -y \
+    && opam switch create 4.10.2
+
+# creates symlink for Python 3
+RUN sudo ln -s /usr/bin/python3 /usr/bin/python
+
+# updates PATH variable for opam
+ENV PATH="/home/opam/.opam/4.10.2/bin:$PATH"
+
+# clones pyre-check and installs dependencies
+RUN git clone https://github.com/facebook/pyre-check
```

Comment:
We should use `ADD` to grab files from the local filesystem:
https://docs.docker.com/engine/reference/builder/#add

Note that we probably want to do this in three steps, to prevent misc file changes from causing the unzip and pip install steps from running:
1. Pull in zip file (then unzip)
2. Pull in requirements.txt (then pip install)
3. Pull in remaining files